### ISSUE TITLE
False positive in UnnecessaryAliasExpansion when alias value is the result of a function call

### ISSUE THREAD

**Issue by heydtn** [rrrene/credo#847]
### Environment

* Credo version: 1.5.4
* Erlang/Elixir version (`elixir -v`): 1.11.3
* Operating system: Mac

### What were you trying to do?

When writing a macro, wrapping a module reference in the alias!/1 macro forces it to resolve via the caller's aliases when compiling.

For example,
```elixir
alias alias!(MyModule)
```

and its use in a proof-of-concept which leverages its functionality:
```elixir
defmodule MyMacro do
  defmacro my_macro do
    quote do
      defmodule MySubmodule do
        alias alias!(MyAliasedModule)

        def test do
          IO.puts(MyAliasedModule)
        end
      end
    end
  end

  defmacro my_dsl(do: block) do
    quote do
      alias __MODULE__.MyAliasedModule

      defmodule MyAliasedModule do
      end

      unquote(block)
    end
  end
end

defmodule MyModule do
  import MyMacro

  my_dsl do
    my_macro
  end
end
```

will result in `MyModule.MySubmodule.test/0` outputting `Elixir.MyModule.MyAliasedModule` as opposed to `Elixir.MyAliasedModule` had `alias!` not been used.

### Expected outcome

Credo to not flag an UnnecessaryAliasExpansion styling error

### Actual outcome

Using both `alias` and `alias!`, credo misconstrues it with an unnecessary alias expansion as it matches the pattern used by the checker.  This can theoretically happen for other function calls as well in the form of `alias my_function(MyModule)` or `alias MyModule.my_function(MyModule2)`.

For `alias alias!(MyAliasedModule)`, the generated AST looks like:
```elixir
{:alias, [context: Elixir],
 [
   {:alias!, [context: Elixir, import: Kernel],
    [{:__aliases__, [alias: false], [:MyAliasedModule]}]}
 ]}
```

which matches the pattern used for checking UnnecessaryAliasExpansion

```elixir
{:alias, _, [{_, _, [{:__aliases__, opts, [child]}]}]}
```

UnnecessaryAliasExpansion may need to specify that the first element of the tuple is a `:.` operation with the second/last argument being a `:{}` to specify that it's an alias expansion.

```elixir
{:alias, _,
  [{
    {:., _, [_, :{}]},
    _,
    [{:__aliases__, opts, [child]}]
  }]
}
```

**Comment by rrrene** [comment]
Thx for reporting! :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by rrrene** [comment]
This is part of v1.6.1 💚
